### PR TITLE
feat: add support for regulation from features.json

### PR DESCRIPTION
**Pull request by koladilip**
# Description
Read regulation destinations from features.json directly, this change has been done with backward compatibility in mind so releases need not be coordinated with transformer

## Linear Ticket

https://linear.app/rudderstack/issue/INT-1336/add-support-for-regulation-in-featuresjson
Resolves INT-1336

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: services/transformer/features.go [hunk 1]
```diff
@@ -31,7 +34,8 @@ var defaultTransformerFeatures = `{
 	"routerTransform": {
 	  "MARKETO": true,
 	  "HS": true
-	}
+	},
+	"regulation": ["AM"],
```
Comment (chandumlg) [inline services/transformer/features.go:38]: Why put this in default?

File: services/transformer/features.go [hunk 2]
```diff
@@ -53,6 +57,11 @@ func NewNoOpService() FeaturesService {
 
 type noopService struct{}
 
+// Regulation implements FeaturesService.
```
Comment (chandumlg) [inline services/transformer/features.go:60]: Delete the comment?

File: services/transformer/features_impl.go [hunk 3]
```diff
@@ -42,6 +43,16 @@ func (t *featuresService) RouterTransform(destType string) bool {
 	return gjson.GetBytes(t.features, "routerTransform."+destType).Bool()
 }
 
+func (t *featuresService) Regulation() []string {
```
Comment (chandumlg) [inline services/transformer/features_impl.go:46]: A test for this function would be good.

**Review by chandumlg — APPROVED** [review]

File: services/transformer/features.go (hunk 1 above)
Comment (koladilip) [reply services/transformer/features.go:38]: Just thought of keeping something like previously, what do you suggest?

File: services/transformer/features.go (hunk 2 above)
Comment (koladilip) [reply services/transformer/features.go:60]: Done

**Review by cisse21 — APPROVED** [review]

**Pull request merged**